packages/drivers/odsp-driver/src/odspDocumentDeltaConnection.ts

```diff
@@ -127,8 +127,7 @@ class SocketReference extends TypedEventEmitter<ISocketEvents> {
             // comes in from "disconnect" listener below, before we close socket.
             this.isPendingInitialConnection = false;
 
-            // Explicitly cast error to the specified event args type to ensure type compatibility
-            this.emit("server_disconnect", error as IFluidErrorBase & OdspError);
+            this.emit("server_disconnect", error);
```

Comment:
Why? I added this because `this.emit` doesn't give type safety on the parameters, so I cast to the type expected by handlers (where we do have type safety).  I guess one drawback is maintainability.  You are welcome to remove it, just wanted to explain why it was there to begin with :)

Reply:
createOdspNetworkError return type is: IFluidErrorBase & OdspError. So, this case is unnecessary here. Lint also complains about it.
Previously it was ok when return type also included IFacetCodes .